src/AppProviders.tsx

```diff
@@ -86,22 +87,24 @@ export function AppProviders({ children }: ProvidersProps) {
                             <KeepKeyProvider>
                               <WalletConnectV2Provider>
                                 <ModalProvider>
-                                  <ErrorBoundary
-                                    FallbackComponent={ErrorPage}
-                                    onError={handleError}
-                                  >
-                                    <TransactionsProvider>
-                                      <AppProvider>
-                                        <FoxEthProvider>
-                                          <DefiManagerProvider>
-                                            <RFOXProvider>
-                                              <FoxPageProvider>{children}</FoxPageProvider>
-                                            </RFOXProvider>
-                                          </DefiManagerProvider>
-                                        </FoxEthProvider>
-                                      </AppProvider>
-                                    </TransactionsProvider>
-                                  </ErrorBoundary>
+                                  <Suspense fallback={defaultSuspenseFallback}>
```

Comment:
So I guess we have to discuss this together as we're not sure to want an app level boundary?

I'm agnostic on this choice myself as I didn't meet any downside of both, we could revert if we feel like it sucks

Reply:
What we've discussed is [error boundaries](https://legacy.reactjs.org/docs/error-boundaries.html). This is a [suspense](https://react.dev/reference/react/Suspense) boundary.